### DISCUSSION TITLE
屏蔽搜索广告建议

### DISCUSSION THREAD

**Discussion by AzusaHana** [the1812/Bilibili-Evolved#3969] [功能建议 / Ideas]
当前部分搜索结果中存在广告，插件【删除广告】版本：3397e56dc，可以将其删除并留下空白

但当前空白位置仍然能够点击跳转到广告页面，是否能去除掉该跳转？
![image](https://user-images.githubusercontent.com/100906636/217271020-c8661ba0-f466-4d66-b8af-d24b4bd560d4.png)
点击会跳转
![image](https://user-images.githubusercontent.com/100906636/217271153-221c00a6-8d72-42e7-949e-12b0d6cb7957.png)
广告链接：[https://cm.bilibili.com/...](https://cm.bilibili.com/cm/api/fees/pc/sync/v2?msg=a%7C5647%2Cb%7Cbilibili%2Cc%7C1%2Cd%7C%2Ce%7CCJmPqAEQpr2TBRjUpaMxIAQoATDWxzo4jyxCHzE2NzU3Nzk1OTE1NjRxMTcyYTI1YTIzYTE1M3E2NDdIjJv94eIwUgbpmLPmsZ9aBuW5v%2BS4nGIG5Lit5Zu9aAFwAHiAgICAoDmAAQOIAQCSASYyNDBlOjNiMTpjMDkxOjE5MzA6MmFkMToyN2ZmOmZlOTk6ZWFmZpoBAKABBKgBzAGyASAUuttDmJEMNXbL4O6jq%2F31dlZ3S0mKE%2B0my1lOoa4A7roBpAJodHRwczovL3JlZGlyLndnZ2FtZXMuY24vcHA1b3U4MnYvP3B1Yl9pZD13b3RfYmlsaV9odWFfOTI1JnhpZD1iaWxpL2JpbGktaHVhLzkyNS8vd290JnRyYWNraWQ9cGJhZXMuVmI0b3FvRnlDZmhyd05LdlRBOUZkWlZrWG16aVI0TGo5TzJqclZ0OXJQSVBjLTRKYWdYYWVZX001TWpJTG5ESENFRm5oNVlLVmFjaWxuYjZ5YWNHdXV6cWhTbF9qS05za3A3TGI3MDd5NjByRHNCM3l6dkZaS3k4c2htNjZHaG5BOVY5UE1YOEZxM1lBelFiOTBpRENQY0hmNHkybWZCdEFpVTlvNERNLUtoNFlfeXZNMEdMcVBMSEdJVlk4NHZ4wgECMzDSAQDYAcoD4AGAlOvcA%2BgBoMIe8AEA%2BAEFgAICiAIAkgK8AjEwNTI1MTYwXzE2NzUyMTg0NTMsMTA0NDQyOTZfMTY3NTU3MjU3Nyw5MDA4OTE3XzE2NzU1NzI1OTAsMTAxNDA4MjZfMTY3NTU3MjYwNSwxMDY4OTU5Ml8xNjc1NTcyNjA3LDEwNjUxOTYzXzE2NzU1NzI2MDUsMTA2NTM4MjRfMTY3NTU3MjYwNywxMDcyOTk4Nl8xNjc1NTcyNjA4LDEwNjc0MjQ4XzE2NzU1NzI2MDgsMTA3MjUyNDRfMTY3NTU3MjYyNyw4Njc4NzA0XzE2NzU1NzI2MjcsMTA3MjU0MjBfMTY3NTU3MjY4OCwxMDcxODM3N18xNjc1NTcyNjg4LDEwNjUxMzk2XzE2NzU1NzI3MDgsMTAzODk3MTBfMTY3NTU3MjcwOCw5NDQzNTM5XzE2NzU2NjgwMDKYAoQIoAIBqAKIBbACALgCAMACgLUYyAIG6gIA%2BAKgAYgDAZIDAKgDALADALgDAMgDANIDjAF7IjEiOiIxMDMzMzg3MDgiLCIxMiI6IjU2NDciLCIxMyI6IjE0MDQ2IiwiMTQiOiI4NTYiLCIxNSI6Ijg3MyIsIjE2IjoiMTQwNDZfMTYwIiwiMiI6Ijk1OTQ0NiIsIjMiOiI5NTk0NDYiLCI0IjoiMzM0IiwiNSI6IjMzOCIsIjYiOiI5NTk0NDYifeADAugDAvADAPoDBW90aGVyggQJbnVsbDpudWxsiAQEkAQAmAQBoAQiqgQHCPWRsQIQBKoEBwju3PICEAG4BArABATQBADYBADiBG97InBzSWQiOjcwMDcsInYyIjoiQVJ2M09NQkdnbFQ2aEphY1lBcjdiOE5VZGF1Q29fUlZuSVppWUlQR0tkRERuSmFKbjBWd2xNYUFGV0FBS1BxcEROSmN5STFnMW90SUsyN2t6TVBvaW1nY3BBIn3oBADwBAD6BNoBeyJiaXpfdHlwZSI6NCwiY3BhIjoie1wiY3BhX2xldmVsXCI6MixcImNwYV9zZXRcIjo2NTAwfSIsImV4cGxvcmVfcmF0aW8iOiIxMDAiLCJtb2RlbFNjb3JlIjoie1wiY3RyXCI6XCIyMDQuNzg0MzYzXCIsXCJjdnJfbDBcIjpcIjYuMzk1MDkyXCIsXCJmaW5hbF9wY3RyXCI6XCIyMDQuNzg0MzYzXCIsXCJmaW5hbF9wY3ZyXCI6XCI2LjM5NTA5MlwifSIsInZpZGVvX3VwX21pZCI6MH2ABQCQBSiQBTOQBUOQBUiQBUmQBWKQBXKQBYQBkAWGAZAFiAGQBYkBkAWsAZAFsQGQBbMBkAW0AZAFyAGQBdABkAXRAZAF1AGQBdcBkAXYAZAF6AGQBfEBkAX5AZAFhQKQBYkCkAWOApAFmAKQBZoCkAWdApAFngKQBakCkAWqApAFswKQBbcCkAXAApAFwgKQBcMCkAXFApAFywKQBcwCkAXOApAF0wKQBdYCkAXXApAF2AKQBdkCkAXaApAF2wKQBdwCkAXfApAF4QKQBeMCkAXkApAF5QKQBeYCkAXwApAF9QKQBfsCkAWGA5AFjQOQBZIDkAWYA5AFmQOQBZoDkAWbA5AFnAOQBaEDkAWiA5AFpAOQBaUDkAWnA5AFsAOQBbEDkAW2A5AFtwOQBcADkAXBA5AFzQOQBdMDkAXXA5AF8gOQBfUDkAX3A5AF%2FwOQBasEkAWvBJAFsASQBbEEkAWyBJAFtASQBbgEoAUAuAUEwAXebQ%3D%3D%2Cf%7Cclick_sync_3%2Cg%7C1%2Ch%7C%2Ci%7C11918382%2Cj%7C%2Ck%7C1675779593649%2Cl%7C5646%2Cm%7C1675760769554%2Cn%7C%2Co%7C103338708%2Cp%7Cad_card&ts=1675779593649&spm_id_from=..search-card.all.click)

**Comment by Moenupa** [comment]
注入一个css应该就行，怕被覆盖就加个`!important`

```js
const BILIBILI_AD_DOMAIN = "//cm.bilibili.com/"
GM_addStyle(`a[href^='${AD_DOMAIN}'] {pointer-events: none; cursor: default;}`)
```

**Reply by the1812** [reply]
这个思路不错

**Reply by Moenupa** [reply]
保险起见可以用 `*=` 怕b站改代码
我之前看dom只发现了 `//cm.bilibili.com/....` 的href，为了些许的性能自己用了prefix match
用 `a[href*='cm.bilibili.com']` 更彻底

**Comment by Geeyun-JY3** [comment]
#3862、#3997
